cip-builtin-drop/README.md

```diff
@@ -0,0 +1,140 @@
+### Function definition
+We define a new Plutus built-in function with the following type signature:
+```haskell
+builtinDropList :: BuiltinInteger -> BuiltinList a -> BuiltinList a
+```
+
+Similar to the behavior of the `indexOfByteString` builtin, this new builtin will simply error if the provided index is out of bounds for the list.
+
+
+### Cost Model
+Although the `BuiltinList` type is a recursive data-type, costing should be relatively straightforward. 
+We propose to define a cost model linear in the size of `n`, the number of elements to drop. What remains is to find a proper coefficient and offset for that linear model, which should be quite easy. 
+
+
+## Rationale: how does this CIP achieve its goals?
+<!-- The rationale fleshes out the specification by describing what motivated the design and what led to particular design decisions. It should describe alternate designs considered and related work. The rationale should provide evidence of consensus within the community and discuss significant objections or concerns raised during the discussion.
+
+It must also explain how the proposal affects the backward compatibility of existing solutions when applicable. If the proposal responds to a CPS, the 'Rationale' section should explain how it addresses the CPS, and answer any questions that the CPS poses for potential solutions.
+-->
+* Easy to implement as it reuses existing code of the Plutus codebase;
+* The built-in is generic enough to cover a wider set of use-cases;
+* The built-in is still relevant even if we get constant lookup index data-structures since there are occasions where BuiltinList would be preferred;
+* This directly addresses the big performance bottleneck that the fixed-point recursion implementation of `elemAt` and `drop` impose on many DApps;
+
+### Alternatives 
+
+- We could decide to accept the heuristic `elemAtFast` implementation as  as an adequate solution.
+- We could provide a more generic builtin that applies a function recursively `n` times (seems complicated and bad idea). 
+- We could try to reduce the overhead introduced by aspects of the `elemAt` by making the language / compiler more performant (still can't imagine we would be able to get anywhere near the performance of this builtin).
+
+## Path to Active
+
+### Acceptance Criteria
+<!-- Describes what are the acceptance criteria whereby a proposal becomes 'Active' -->
+- [] Fully implemented in Cardano.
+      
+### Implementation Plan
+<!-- A plan to meet those criteria. Or `N/A` if not applicable. -->
+- [] Passes all requirements of both Plutus and Ledger teams as agreed to improve Plutus script efficiency and usability.
```

Comment:
@michaelpj please let us know if this single checkbox item properly signifies all the requirements for Plutus Core changes as per CIP-0035, or if you think these need to be itemised further here & in similar proposals.